lms/static/scripts/postmessage_json_rpc/server/server.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+export default class Server {
+  constructor(window_) {
+    window_ = window_ || window;
+    window_.addEventListener(
+      "message",
+      (event) => this._receiveMessage(event),
```

Comment:
The `this` context will be lost in `this._receiveMessage`, unlike `self._aMethod` in Python. I suggest binding `this` or using a lambda.

Reply:
This seems to work correctly - `this` inside of `receiveMessage` is the `Server` instance. Because I passed `(event) => this._receiveMessage(event)` rather than just passing `this._receiveMessage` directly. Is there a more idiomatic way to bind `this`?

Reply:
Oh right, sorry. This code is fine. You could also use bind (`this._receiveMessage(event).bind(this)` instead of passing an arrow function) but I don't see any advantage in this context.